GeoServer 3.0.0 Release (#319)
geoserver/geoserver.github.io: _posts/2026-06-11-geoserver-3-0-0-released.md

---
author: Jody Garnett
date: 2026-06-11
layout: post
title: GeoServer 3.0.0 Release
categories:
- Announcements
- Vulnerability
tags:
- Release
release: release_30
version: 3.0.0
jira_version: 17404
--- 

GeoServer [3.0.0](/release/3.0.0/) release is now available
with downloads
([bin](https://sourceforge.net/projects/geoserver/files/GeoServer/3.0.0/geoserver-3.0.0-bin.zip/download),
[war](https://sourceforge.net/projects/geoserver/files/GeoServer/3.0.0/geoserver-3.0.0-war.zip/download),
[windows](https://sourceforge.net/projects/geoserver/files/GeoServer/3.0.0/GeoServer-3.0.0-winsetup.exe/download)), along with 
[docs](https://sourceforge.net/projects/geoserver/files/GeoServer/3.0.0/geoserver-3.0.0-htmldoc.zip/download) and
[extensions](https://sourceforge.net/projects/geoserver/files/GeoServer/3.0.0/extensions/).

> This is a stable release of GeoServer recommended for production use.

Reviewer: While we did everything possible to make it a very solid release, I'm not confident recommending it for production usage yet, due to the massive number of changes.

Maybe:

```suggestion
This is the first stable release of GeoServer 3.x series.
```

GeoServer 3.0.0 is made in conjunction with GeoTools 35.0, and GeoWebCache 2.0.0. 

Thanks to Jody Garnett (GeoCat), and Peter Smythe (AfriGIS) for making this release. 

## Security Considerations

This release addresses security vulnerabilities and is an important upgrade for production systems.

See project [security policy](https://github.com/geoserver/geoserver/blob/main/SECURITY.md) for more information on how security vulnerabilities are managed.

## Welcome to GeoServer 3

> We are overjoyed to share the release of GeoServer 3.0 with our community, this is the final stretch of a long road, a year of development, and a lot of planning and support to make it all happen. Thanks to all the organizations and individuals supporting GeoServer 3.

Reviewer 2: I'd stick with either "GeoServer 3" or GeoServer 3.0.0"

```suggestion
We are overjoyed to share the release of GeoServer 3 with our community, this is the final stretch of a long road, a year of development, and a lot of planning and support to make it all happen. Thanks to all the organizations and individuals supporting GeoServer 3.
```

<a href="/img/posts/3.0/welcome-global.png" target="_blank" rel="noopener">
  <img src="/img/posts/3.0/welcome-global.png" alt="GeoServer 3" class="screensnap"
     style="max-width: 95%"/>
</a>

### Straightforward upgrade

> We have taken great pains to make the upgrade process seamless from GeoServer 2.28.x.

Reviewer: I would be a bit more positive here:

```suggestion
Special care has been taken to ensure a seamless upgrade from GeoServer 2.28.x.
```

1. Important: We have made **no changes** to the GeoServer Data Directory.
   

>    Download and try GeoServer 3.0-RC today!

Reviewer 3: Not RC

>    Download and try GeoServer 3.0-RC today!

Reviewer 2: ```suggestion
   Download and try GeoServer 3.0.0 today!
```

  <a href="/img/posts/3.0/menus-responsive.png" target="_blank" rel="noopener">
    <img src="/img/posts/3.0/menus-responsive.png" alt="Responsive Theme: Menus" class="screensnap"
     style="max-width: 50%"/>
  </a>

* **Forms**: Forms have adopted a two-column layout adapting to page width.
  
  <a href="/img/posts/3.0/form-two-column.png" target="_blank" rel="noopener">
    <img src="/img/posts/3.0/form-two-column.png" alt="Responsive Theme: Form two-column layerout" class="screensnap"
     style="max-width: 50%"/>
  </a>

Thanks to Stefano Bovio (GeoSolutions) for leading this frequently requested improvement, the entire GeoServer 3 team for implementing and checking, and testers at AfriGIS and GeoCat for verifying and updating screenshots.

### New Layer Preview

A new full-screen layer preview is provided using the latest OpenLayers library.

<a href="/img/posts/3.0/ol-preview.png" target="_blank" rel="noopener">
  <img src="/img/posts/3.0/ol-preview.png" alt="New full screen layer preview" class="screensnap"/>
</a>

Thanks to Stefano Bovio (GeoSolutions) for the welcome improvement.

## Updated Environment

> GeoServer 3 is overjoyed to support Tomcat 11.0.x and Jetty 12.1 application servers after completing our transition to Spring Framework 7 and Jakarta EE Servlet API 6.1.

Reviewer: GeoServer is not a person LOL (and those are requirement, not a nice extra)

```suggestion
GeoServer 3 requires Tomcat 11.0.x and Jetty 12.1 application servers after completing our transition to Spring Framework 7 and Jakarta EE Servlet API 6.1.
```

| GeoServer 2.28.x | 17, 21 |              | Jetty 9.4  | Servlet API 3.1 |                |
{: .compat-table }

For more information see [container considerations](https://docs.geoserver.org/latest/en/user/production/container/).

Thanks to the entire GeoServer 3 team and [crowdfunding campaign](/sponsor/gs3-crowdfunding) for this major accomplishment, representing the completion of Milestone 3.

## OAuth2 OpenID Connect Extension

The new OAuth2 OpenID Connect Security Integration (OIDC) plugin is now an official extension.

The transition to Spring Security 7 was one of the big tasks accomplished for GeoServer 3. This work includes the creation of a new `OIDC` plugin. The new plugin has taken over the responsibilities of previously available `Keycloak` and `OAuth2` plugins.

For guidance on upgrading please see the detailed  [migration guide](https://docs.geoserver.org/main/en/user/extensions/oidc/migrating/).

Thanks to Alessio Fabiani and others for this important improvement. Special thanks to everyone who provided feedback and testing during the 3.0-RC timeframe, your success has allowing this module to graduate to full extension for 3.0.0 release.

## New Documentation

> The long-awaited transition to Markdown documentation has finally arrived. Welcome to our new [User Manual](https://docs.geoserver.org/latest/en/user/).  The older GeoServer 2.x documentation is available at [Docs Archive](https://docs-archive.geoserver.org/) or via the version switcher.  Please help out by fixing any remaining [small issues](https://docs.geoserver.org/latest/en/docguide/quickfix/) or log an issue for Peter to address.

Reviewer: The 2.x documentation series is actually easily available through the version selector in the documentation web site:

<img width="687" height="232" alt="Image" src="https://github.com/user-attachments/assets/968208c1-b52e-4bc9-9acc-23df07db849c" />

Yes, that points eventually to the archive site, for sure, but we don't need to make it sounds like a dead end, 2.28.x is still going to be released and we need to publish its documentation accordingly no?

<a href="https://docs.geoserver.org/main/en/user/">
  <img src="/img/posts/3.0/user-manual.png" alt="The new user manual" class="screensnap"/>
</a>

Thanks to Peter Smythe (AfriGIS) and Jody Garnett (GeoCat) for working on this activity which ended up being an incredible amount of work.

### Pending Community Modules

> THe documentation contains a new heading for [pending community modules](https://docs.geoserver.org/main/en/user/community/#pending-community-modules) that are seeking public use and support in order

Reviewer: ```suggestion
The documentation contains a new heading for [pending community modules](https://docs.geoserver.org/main/en/user/community/#pending-community-modules) that are seeking public use and support in order
```

Given the graduation of OpenID connect, and OSEO being kicked out of the build, do we actually have any pending module?

Author: We do not, I nominated jwt-headers but nobody responded.

Author: Is OSEO coming back to the build?

Reviewer: Likely so, but not today
